src/includes/classes/Actions.php

```diff
@@ -798,17 +798,23 @@ protected function proUpdate($args)
         ]);
         $this->plugin->dismissMainNotice('new-pro-version-available');
 
-        $redirect_to = self_admin_url('/update.php');
-        $query_args  = [ // Like a normal WP plugin.
-                         'action'   => 'upgrade-plugin',
-                         'plugin'   => plugin_basename(PLUGIN_FILE),
-                         '_wpnonce' => wp_create_nonce('upgrade-plugin_'.plugin_basename(PLUGIN_FILE)),
-
-                         // See: `preSiteTransientUpdatePlugins()` where these are picked up.
-                         GLOBAL_NS.'_update_pro_version' => apply_filters(GLOBAL_NS.'_update_pro_version', $product_api_response->pro_version),
-                         GLOBAL_NS.'_update_pro_zip'     => base64_encode(apply_filters(GLOBAL_NS.'_update_pro_zip', $product_api_response->pro_zip)),
-                         // @TODO Encrypt/decrypt to avoid mod_security issues. Base64 is not enough.
+        $update_pro_version = apply_filters(GLOBAL_NS.'_update_pro_version', $product_api_response->pro_version);
+        $update_pro_zip     = apply_filters(GLOBAL_NS.'_update_pro_zip', $product_api_response->pro_zip);
+        $redirect_to        = self_admin_url('/update.php');
+        $query_args         = [ // Like a normal WP plugin.
+                                'action'                        => 'upgrade-plugin',
+                                'plugin'                        => plugin_basename(PLUGIN_FILE),
+                                '_wpnonce'                      => wp_create_nonce('upgrade-plugin_'.plugin_basename(PLUGIN_FILE)),
+                                GLOBAL_NS.'_update_pro_version' => $update_pro_version,
+                                GLOBAL_NS.'_update_pro_zip'     => base64_encode(apply_filters(GLOBAL_NS.'_update_pro_zip', $product_api_response->pro_zip)),
         ];
+
+        // Store the Pro Zip URL in Transient with a 5 minute expiration
+        // See: `preSiteTransientUpdatePlugins()` where these are picked up.
+        if (false === (get_site_transient(GLOBAL_NS.'_update_pro_zip_'.$update_pro_version))) {
```

Comment:
Also, I wouldn't worry about checking if it exists here, because this routine is designed specifically to update those values with the latest details that it just pulled from the API. So I'd skip the `get_site_transient()` call altogether here, and just set the transient value, which will update it if it already exists.